container/image.bzl

```diff
@@ -244,35 +194,41 @@ def _impl(ctx, files=None, file_map=None, empty_files=None, directory=None,
   cmd = cmd or ctx.attr.cmd
   symlinks = symlinks or ctx.attr.symlinks
   output = output or ctx.outputs.executable
-  env = env or ctx.attr.env
+  layers = container_layers or ctx.attr.container_layers
   debs = debs or ctx.files.debs
   tars = tars or ctx.files.tars
 
   # Generate the unzipped filesystem layer, and its sha256 (aka diff_id).
-  unzipped_layer, diff_id = _build_layer(ctx, files=files, file_map=file_map,
-                                         empty_files=empty_files,
-                                         directory=directory, symlinks=symlinks,
-                                         debs=debs, tars=tars)
+  unzipped_layer, diff_id = build_layer(ctx, files=files, file_map=file_map,
+                                        empty_files=empty_files,
+                                        directory=directory, symlinks=symlinks,
+                                        debs=debs, tars=tars)
 
   # Generate the zipped filesystem layer, and its sha256 (aka blob sum)
-  zipped_layer, blob_sum = _zip_layer(ctx, unzipped_layer)
+  zipped_layer, blob_sum = zip_layer(ctx, unzipped_layer)
+
+  # Get the layers and shas from our base.
+  # These are ordered as they'd appear in the v2.2 config,
+  # so they grow at the end.
+  parent_parts = _get_layers(ctx, ctx.attr.base, ctx.files.base)
+  zipped_layers = parent_parts.get("zipped_layer", []) + _getLayerInfo(layers, "zipped_layer") + [zipped_layer]
+  shas = parent_parts.get("blobsum", [])  + _getLayerInfo(layers, "blob_sum") + [blob_sum]
+  unzipped_layers = parent_parts.get("unzipped_layer", []) + _getLayerInfo(layers, "unzipped_layer") + [unzipped_layer]
+  layer_diff_ids = _getLayerInfo(layers, "diff_id") + [diff_id]
```

Comment:
nit: You are inconsistent about whether this portion is hoisted (like this one) or a nested expression.  IMO pick one and keep these lines more consistent.

Reply:
Sorry that I didn't quite get the the meaning of "hoisted". 

Do you mean I should use all nested expression e.g. list comprehension in this case?

Reply:
By hoisted I mean:
```
foo = _getLayerInfo(...)
bar = parent_parts.get(...) + foo
```
vs:
```
bar = parent_parts.get(...) + _getLayerInfo(...)
```